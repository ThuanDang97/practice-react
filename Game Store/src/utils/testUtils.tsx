import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import { BrowserRouter as Router } from 'react-router-dom'

// Themes
import theme from '@self-types/Themes'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <AuthProvider>
        <CartProvider>
          <Router>{component}</Router>
        </CartProvider>
      </AuthProvider>
    </ThemeProvider>,
  )
}

export * from '@testing-library/react'
export { customRender as render }
