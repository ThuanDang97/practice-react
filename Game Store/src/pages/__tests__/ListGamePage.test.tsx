// Utils
import { render } from '@utils/testUtils'

// Page
import ListPage from '@pages/index'

describe('List Game Page', () => {
  beforeEach(() => {
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'),
      useParams: jest.fn(() => ({
        name: 'GTA',
      })),
      useRouteMatch: () => ({ url: '' }),
    }))
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  it('should render CList Game Page', () => {
    const { container } = render(<ListPage />)

    expect(container).toMatchSnapshot()
  })
})
