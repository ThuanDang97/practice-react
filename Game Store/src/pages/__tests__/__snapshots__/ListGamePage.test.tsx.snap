// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Game Page should render CList Game Page 1`] = `
<div>
  <div
    class="main MuiBox-root css-1n0bnyw"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-1ybknf5-MuiTypography-root"
    >
      Best of All Time
    </h1>
    <div
      class="infinite-scroll-component__outerdiv"
    >
      <div
        class="infinite-scroll-component "
        style="height: auto; overflow: auto;"
      >
        <div
          class="MuiBox-root css-1ezr6d7"
        />
        <h1
          class="MuiTypography-root MuiTypography-h1 css-dl3eaz-MuiTypography-root"
        >
          Loading...
        </h1>
      </div>
    </div>
  </div>
</div>
`;
